template.yml.erb

```diff
@@ -41,6 +41,146 @@ Globals:
 Conditions:
   IsDevCondition: !Equals [!Ref BaseDomainName, "dev-code.org"]
 Resources:
+  HttpDomain:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      HostedZoneName: !Sub "${BaseDomainName}."
+      Name: !Sub "${SubDomainName}-http.${BaseDomainName}"
+      Type: A
+      AliasTarget:
+        DNSName: !GetAtt HttpDomainName.RegionalDomainName
+        HostedZoneId: !GetAtt HttpDomainName.RegionalHostedZoneId
+
+  HttpDomainName:
+    Type: AWS::ApiGatewayV2::DomainName
+    Properties:
+      DomainName: !Sub "${SubDomainName}-http.${BaseDomainName}"
+      DomainNameConfigurations:
+        - EndpointType: REGIONAL
+          CertificateArn: !Ref HttpCertificate
+          CertificateName: !Sub "${SubDomainName}-http.${BaseDomainName}"
+
+  HttpCertificate:
+    Type: AWS::CertificateManager::Certificate
+    Properties:
+      DomainName: !Sub "${SubDomainName}-http.${BaseDomainName}"
+      ValidationMethod: DNS
+      DomainValidationOptions:
+        - DomainName: !Sub "${SubDomainName}-http.${BaseDomainName}"
+          HostedZoneId: !Ref BaseDomainNameHostedZonedID
+
+  HttpDomainNameAPIMapping:
+    Type: AWS::ApiGatewayV2::ApiMapping
+    DependsOn:
+      - HttpDomain
+    Properties:
+      ApiId: !Ref HttpAPI
+      DomainName: !Sub "${SubDomainName}-http.${BaseDomainName}"
+      Stage: !Ref HttpStage
+
+  HttpAPI:
+    Type: AWS::ApiGatewayV2::Api
+    Properties:
+      Name: !Sub "${SubDomainName}-http.${BaseDomainName}"
+      ProtocolType: HTTP
+
+  PutRoute:
+    Type: AWS::ApiGatewayV2::Route
+    Properties:
+      ApiId: !Ref HttpAPI
+      RouteKey: PUT /seedsources/sources.json
```

Comment:
will the final POST/PUT url be `https://javabuilderbeta-http.code.org/seedsources/sources.json`? We may not need the `sources.json` part since we already assign the file name directly in the lambda, right?

Reply:
Yeah, we can make the latter part of the URL whatever we want. I can also update this to be 
`PUT /seedsources/{filename}`
instead, so we have more flexibility on the filename. Thoughts?

Reply:
I think we'll want to be restrictive on the file name because Javabuilder is currently configured to always look for a file named "sources.json". In which case either `PUT /seedsources` or `PUT /seedsources/sources.json` should be fine.